sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1188,6 +1188,12 @@ class Dataset[T] private[sql](
   def col(colName: String): Column = colName match {
     case "*" =>
       Column(ResolvedStar(queryExecution.analyzed.output))
+    case ParserUtils.escapedIdentifier(columnNameRegex)
```

Comment:
no. this improves the current behavior. when it is `a`, if a is the column name, it will expand to just column a. if it is `(a)?+.+`, it will be treated it as regular expression and expand. (the current behavior is to throw AnalysisException.